test-basic/concurrent-requests.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var testconfig = require('../etc/test-config.js');
+
+var marklogic = require('../');
+
+var dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+
+describe('concurrent-requests test', function(){
+
+
+    it('should write 1000 documents concurrently', function(done){
+        for(var i=0; i<3; i++) {
+            writeBatch(i,0);
+        }
+
+        done();
+    });
+});
+
+
+function writeBatch(writerId, nextDocument) {
+    if(nextDocument>=1000) {
+        console.log(writerId + ' is done');
+        return;
+    }
+    var documentDescriptors = createDocumentDescriptors();
```

Comment:
The documentDescriptors array and the nextDocument counter should be shared across all calls to writeBatch().

I'd suggest turning the documentDescriptors variable into a module-level constant by moving the declaration immediately before the start of the writeBatch() function on line 36.

Also, I was mistaken to suggest that nextDocument should be a parameter.  I'd suggest turning it into a module-level variable.

Putting the two suggestions together, the declaration would look like:

    const documentDescriptors = createDocumentDescriptors();
    let nextDocument = 0;
    function writeBatch(writerId) {
        ...
    }

By the way, the var keyword is discouraged in recent versions of JavaScript because it is not properly scoped in execution.  The recommendation is to use const to declare immutable constants (equivalent to final in Java) and let to define mutable variables.